File: test/test_vector_disable.py
# Python module test_vector_disable.py
#
# Copyright (c) 2003 Raoul M. Gough
#
# Use, modification and distribution is subject to the Boost Software
# License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy
# at http://www.boost.org/LICENSE_1_0.txt)
#
# History
# =======
# 2003/11/19   rmg     File creation
#
# $Id$
#

'''>>> from test_vector_disable_ext import *
>>> v = Vector_disable()
>>> v.append (1)
>>> v.append (2)
>>> v.append (3)
>>> v.append (4)
>>> del v[0]
>>> print [ x for x in v ]
[2, 3, 4]
>>> v[0] = 5
>>> print [ x for x in v ]
[5, 3, 4]
>>> assert getattr (v, "len", None) == None
>>> assert getattr (v, "sort", None) == None
>>> assert getattr (v, "reverse", None) == None
>>> assert getattr (v, "index", None) == None
>>> assert getattr (v, "count", None) == None
>>> assert getattr (v, "insert", None) == None
>>> assert getattr (v, "extend", None) == None
>>> try:
...     s = v[:]
...     print "__getitem__ supports slices (should be disabled)"
... except TypeError, e:
...     print "Got expected exception"
...
Got expected exception
>>> try:
...     v[:1] = [1]
...     print "__setitem__ supports slices (should be disabled)"
... except TypeError, e:
...     print "Got expected exception"
...
Got expected exception
>>> try:
...     del v[:]
...     print "__delitem__ supports slices (should be disabled)"
... except TypeError, e:
...     print "Got expected exception"
...
Got expected exception
'''

def run(args = None):
    import sys
    import doctest

    if args is not None:
        sys.argv = args
    return doctest.testmod(sys.modules.get(__name__))

if __name__ == '__main__':
    print 'running...'
    import sys
    status = run()[0]
    if (status == 0): print "Done."
    sys.exit(status)

File: test/test_vector_disable.cpp
// Copyright (c) 2003 Raoul M. Gough
//
// Use, modification and distribution is subject to the Boost Software
// License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy
// at http://www.boost.org/LICENSE_1_0.txt)
//
// Module test_vector_disable.cpp
//
// Expose a vector with minimal supported methods. Should produce a
// much smaller object file that otherwise.
//
// History
// =======
// 2003/11/19   rmg     File creation
//
// $Id$
//

#include <boost/python/suite/indexing/container_suite.hpp>
#include <boost/python/suite/indexing/vector.hpp>
#include <vector>
#include <boost/python/class.hpp>
#include <boost/python/module.hpp>
#include <boost/python/def.hpp>

BOOST_PYTHON_MODULE(test_vector_disable_ext)
{
  namespace indexing = boost::python::indexing;

  typedef std::vector<int> Container1;

  // Binary mask for the four methods we want to support
  unsigned int const mask
    = indexing::method_getitem
    | indexing::method_setitem
    | indexing::method_delitem
    | indexing::method_append;

#if !defined (BOOST_NO_TEMPLATE_PARTIAL_SPECIALIZATION)
  // Normal case (automatic algorithm and traits selection)
  typedef indexing::container_suite<Container1, mask>
    Suite1;
#else
  // For broken compilers - explicit selection of algorithms/traits
  typedef indexing::vector_suite<Container1, mask>
    Suite1;
#endif

  boost::python::class_<Container1>("Vector_disable").def (Suite1());
}
